library/std/src/panic.rs

```diff
@@ -509,8 +510,16 @@ pub fn get_backtrace_style() -> Option<BacktraceStyle> {
         None if crate::sys::FULL_BACKTRACE_DEFAULT => BacktraceStyle::Full,
         None => BacktraceStyle::Off,
     };
-    set_backtrace_style(format);
-    Some(format)
+
+    match SHOULD_CAPTURE.compare_exchange(
+        current,
```

Comment:
```suggestion
        0,
```
I think this makes the code clearer since 0 is the initial state and nothing ever sets it back to 0.